aws/resource_aws_security_group_rule.go

```diff
@@ -845,3 +872,127 @@ func resourceSecurityGroupRuleDescriptionUpdate(conn *ec2.EC2, d *schema.Resourc
 
 	return nil
 }
+
+// validateSecurityGroupRuleImportString does minimal validation of import string without going to AWS
+func validateSecurityGroupRuleImportString(importStr string) ([]string, error) {
+	// example: sg-09a093729ef9382a6_in_tcp_8000_10.0.3.0/24
+	// example: sg-09a093729ef9382a6_in_92_all_10.0.3.0/24_10.0.4.0/24
+	// example: sg-09a093729ef9382a6_out_tcp_8000_10.0.3.0/24
+	// example: sg-09a093729ef9382a6_out_tcp_8000_pl-34800000
+	// example: sg-09a093729ef9382a6_in_all_all_sg-08123412342323
+	// example: sg-09a093729ef9382a6_in_tcp_100-121_10.1.0.0/16_2001:db8::/48_10.2.0.0/16_2002:db8::/48
+
+	log.Printf("[DEBUG] Validating import string %s", importStr)
+
+	importParts := strings.Split(strings.ToLower(importStr), "_")
+	errStr := "unexpected format of import string (%q), expected SGID_IN/OUT_PROTOCOL_PORT_SOURCE[_SOURCE]*: %s"
+	if len(importParts) < 5 {
+		return nil, fmt.Errorf(errStr, importStr, "too few parts")
+	}
+
+	sgID := importParts[0]
+	ruleType := importParts[1]
+	protocol := importParts[2]
+	portsRaw := importParts[3]
+	sources := importParts[4:]
+
+	if !strings.HasPrefix(sgID, "sg-") {
+		return nil, fmt.Errorf(errStr, importStr, "invalid security group ID")
+	}
+
+	if ruleType != "in" && ruleType != "out" {
```

Comment:
Could you adjust these to `ingress` and `egress` to match the resources and API please?